lib/rubocop/cop/metrics/line_length.rb

```diff
@@ -40,6 +42,45 @@ def max
           cop_config['Max']
         end
 
+        def allow_heredoc?
+          allowed_heredoc
+        end
+
+        def allowed_heredoc
+          cop_config['AllowHeredoc']
+        end
+
+        def extract_heredocs(processed_source)
+          heredocs = []
+          iter = processed_source.tokens.each
+          loop do
+            token = iter.next
```

Comment:
@alexdowad Thanks! I had looked into the AST, but I overlooked that it actually keeps track of the heredoc range. I've pushed another commit, which makes the code significantly easier to follow.